src/main/java/org/opentripplanner/routing/algorithm/strategies/InterleavedBidirectionalHeuristic.java

```diff
@@ -132,42 +223,65 @@ public void initialize(RoutingRequest request, long abortTime) {
      * This function supplies the main search with an (under)estimate of the remaining path weight to the target.
      * No matter how much progress has been made on the reverse heuristic search, we must return an underestimate
      * of the cost to reach the target (i.e. the heuristic must be admissible).
-     * All on-street vertices within walking distance of the origin or destination will have been explored by the
-     * heuristic before the main search starts.
+     * All on-street vertices within walking (or driving) distance of the origin or destination will
+     * have been explored by the heuristic before the main search starts.
      */
     @Override
     public double estimateRemainingWeight (State s) {
         final Vertex v = s.getVertex();
-        if (v instanceof StreetLocation) {
-            // Temporary vertices (StreetLocations) might not be found in the street searches.
+        if (v instanceof StreetLocation || v instanceof TemporaryVertex) {
```

Comment:
I saw you had already helped @abyrd understand this part but could you still elaborate how this check replaces the instanceof checks you had here earlier. For example, BikeRentalStationVertex itself isnt a TemporaryVertex but there is always(?) TemporaryVertex linked to it or how does this work?

Reply:
Here's a diagram of the inheritance of the vertex types that now includes the vertex types introduced from the GTFS-FLEX work.

![otp vertex inheritance 1](https://user-images.githubusercontent.com/3112493/53056974-23964200-3462-11e9-93bb-29e2e0b2d507.png)

After discovering that the GTFS-FLEX vertices inherit from the TemporaryVertex I went ahead and changed the ordering of the vertex type checks in this PR. There are 5 possibilities that need to be checked for. The first is checking if a vertex is an instance of a TransitVertex in which case it should lookup the remaining weight fron the transit vertices. The second checks if a vertex is a StreetLocation or TemporaryVertex which need to be searchable but sometimes are not discovered during the streetSearch method. The third is checking whether the search has not yet boarded transit in which case if a preTransitVertex has been found, the remaining weight should be 0, but not Infinity. Fourthly, a check is needed to see if the search has boarded transit and is near the destination in which case a remaining weight estimate can be returned. And finally if none of those options have been found, the vertex must be a StreetVertex that is not close enough to the origin or destination to warrant further examination.

I hadn't seen Entur's fork and change. I think what this is doing is allowing the streetSearch method to explore nearby streets and thus other stops if the starting vertex happens to be a transit stop.

Reply:
Thanks @evansiroky for adapting to the flex transit changes, and the inheritance diagram is very helpful in understanding the situation.